pkg/cache/factory.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright The Ratify Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cache
+
+import (
+	"context"
+	"fmt"
+)
+
+type CacheFactory interface {
+	Create(ctx context.Context, cacheEndpoint string, cacheSize int) (CacheProvider, error)
+}
+
+var cacheProviderFactories = make(map[string]CacheFactory)
+var memoryCache CacheProvider
+
+// Register adds the factory to the built in providers map
+func Register(name string, factory CacheFactory) {
+	if _, registered := cacheProviderFactories[name]; registered {
+		panic(fmt.Sprintf("cache provider named %s already registered", name))
+	}
+
+	cacheProviderFactories[name] = factory
+}
+
+// NewCacheProvider creates a new cache provider based on the name
+func NewCacheProvider(ctx context.Context, name string, cacheEndpoint string, cacheSize int) (CacheProvider, error) {
```

Comment:
should we consider making a generic interface so that different providers can take in different configs? For example, the ristretto cache store needs to specify the max size values in MB but it doesn't need to specify things like endpoint.